Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -278,11 +278,21 @@ infix operator ++: AdditionPrecedence
 public extension Tensor {
     /// Returns a transposed tensor, with dimensions permuted in the specified order.
     @inlinable
+    @differentiable(
+        wrt: self,
+        vjp: _vjpTransposed(withPermutations:) where Scalar: TensorFlowFloatingPoint)
```

Comment:
Also, rename method `_vjpTransposed(withPermutations:)` to Also, rename `_vjpTransposed(permutations:)`.

```suggestion
        vjp: _vjpTransposed(permutations:) where Scalar: TensorFlowFloatingPoint)
```

Reply:
Would it be also a good idea to use the new `transposed(permutation:)` in methods that use the old one in the same file, to avoid warnings when compiling?

For example,  in `_vjpGathering` 

```swift
let transposedValues = values.transposed(withPermutations: permutations)
```

or overloading methods.

```swift
func transposed(withPermutations permutations: [Int]) -> Tensor {
        let permutations = permutations.map(Int32.init)
        return transposed(withPermutations: Tensor<Int32>(permutations))
}
```

Reply:
Yes.

Reply:
Thanks, all right. It is done.